src/pocketmine/block/EnderChest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\item\Item;
+use pocketmine\item\ItemFactory;
+use pocketmine\item\Tool;
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\Vector3;
+use pocketmine\Player;
+use pocketmine\tile\EnderChest as TileEnderChest;
+use pocketmine\tile\Tile;
+
+class EnderChest extends Transparent{
+
+	protected $id = self::ENDER_CHEST;
+
+	public function __construct(int $meta = 0){
+		$this->meta = $meta;
+	}
+
+	public function getHardness() : float{
+		return 22.5;
+	}
+
+	public function getName() : string{
+		return "Chest";
+	}
+
+	public function getToolType() : int{
+		return Tool::TYPE_PICKAXE;
+	}
+
+	protected function recalculateBoundingBox() : ?AxisAlignedBB{
+		return new AxisAlignedBB(
+			$this->x + 0.0625,
+			$this->y,
+			$this->z + 0.0625,
+			$this->x + 0.9375,
+			$this->y + 0.9475,
+			$this->z + 0.9375
+		);
+	}
+
+	public function place(Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $facePos, Player $player = null) : bool{
+		$faces = [
+			0 => 4,
+			1 => 2,
+			2 => 5,
+			3 => 3
+		];
+
+		$this->meta = $faces[$player instanceof Player ? $player->getDirection() : 0];
+
+		$this->getLevel()->setBlock($blockReplace, $this, true, true);
+		Tile::createTile(Tile::ENDER_CHEST, $this->getLevel(), TileEnderChest::createNBT($this, $face, $item, $player));
+
+		return true;
+	}
+
+	public function onActivate(Item $item, Player $player = null) : bool{
+		if($player instanceof Player){
+
+			$t = $this->getLevel()->getTile($this);
+			$enderChest = null;
+			if($t instanceof TileEnderChest){
+				$enderChest = $t;
+			}else{
+				$enderChest = Tile::createTile(Tile::ENDER_CHEST, $this->getLevel(), TileEnderChest::createNBT($this));
+			}
+
+			if(!$this->getSide(Vector3::SIDE_UP)->isTransparent()){
+				return true;
+			}
+
+			$player->getEnderChestInventory()->openAt($enderChest);
+			$player->addWindow($player->getEnderChestInventory());
+		}
+
+		return true;
+	}
+
+	public function getDrops(Item $item) : array{
+		return [
+			ItemFactory::get(Item::ENDER_CHEST, 0, 8)
```

Comment:
Ender chests drop 8 obsidian. That drop is only applicable if the item has the `Enchantment::SILK_TOUCH` enchant.

Reply:
Yea, I was halfway through looking at other blocks to double check silk touch hadn't been implemented and forgot to change the item id back to obsidian.